tests/test_buffer_manager.py

```diff
@@ -0,0 +1,222 @@
+import numpy as np
+from numpy import testing as npt
+import pytest
+
+import pygfx
+
+from fastplotlib.graphics._features import ColorFeature, PointsDataFeature
+from fastplotlib.graphics._features.utils import parse_colors
+
+
+# TODO: parameterize every test where the color is given in as str, array, tuple, and list
+
+
+def generate_color_inputs(name: str) -> list[str, np.ndarray, list, tuple]:
+    color = pygfx.Color(name)
+
+    s = name
+    a = np.array(color)
+    l = list(color)
+    t = tuple(color)
+
+    return [s, a, l, t]
+
+
+def generate_slice_indices(kind: int):
```

Comment:
@apasarkar do we have other ways of slicing that applies to lines/scatters along the first (n_datapoints, or n_timepoints) direction?